src/apple/macos/component/mod.rs

```diff
@@ -0,0 +1,13 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+#[cfg(target_arch = "x86")]
+pub(crate) mod x86;
+
+#[cfg(target_arch = "x86")]
+pub use self::x86::*;
+
+#[cfg(target_arch = "aarch64")]
```

Comment:
I'm not a big fan of this switch. The one big question is: does this code work on other processors than M1 (and others)? Because if so, we can just replace the existing with yours.

Reply:
I tried it on Intel mac and it does not work

Reply:
Ok, that solves the question then. :)